mycpp/leaky_mylib.cc

```diff
@@ -8,8 +8,18 @@ mylib::FormatStringer gBuf;
 
 namespace mylib {
 
-Str* StrFromBuf(const Buf& buf) {
-  return ::StrFromC(buf.data_, buf.len_);
+Str* StrFromBuf(const Buf* buf) {
+  return ::StrFromC(buf->data_, buf->len_);
+}
+
+Buf* NewBuf(int cap) {
+  void* place = gHeap.Allocate(sizeof(Buf) + cap + 1);
+
+  auto* b = new (place) Buf();
+  b->cap_ = cap;
```

Comment:
style issue: It seems like we could make cap_ a constructor param rather than initializing to zero and then mutating it here

`len_` could also rely on the constructor